### ISSUE TITLE
Kulturen automatisch auf aktiv setzen, wenn Anlieferung in inaktive Kultur

### ISSUE THREAD

**Issue by jmenzi** [barbalex/vermehrung#94]
Um Fehler zu vermeiden, wäre es hilfreich oben beschriebenen Automatismus einzubauen.

**Comment by jmenzi** [comment]
Auch wäre es hilfreich (aber auch etwas gefährlich) Zwischenlager, welche auf inaktiv gesetzt werden automatisch zu leeren.

**Comment by barbalex** [comment]
> Kulturen automatisch auf aktiv setzen, wenn Anlieferung in inaktive Kultur

Du bist dir bewusst: Wenn du in der Lieferung das nach-Feld ausfüllen willst. Und zuerst die falsche Kultur wählst. Und dies dann korrigierst: Dann wurde die zuerst gewählte (falls sie inaktiv war) aktiviert. Und die Chance, dass du das merkst und korrigierst ist nicht besonders gross.

Daher besteht als Nebeneffekt die Gefahr, dass sich Daten verändern, ohne dass man sich dessen bewusst ist. Was dann das Vertrauen in die Daten senkt. Was in meinen Augen viel schlimmer ist, als die kleine Beschleunigung der Datenerfassung.

Die besteht nämlich darin, dass du vermehrung zwei mal offen hast. Und im zweiten Fenster die Kultur aktiv setzt.

Ich habe übrigens schon lange eine Idee, die dir das ohne die Nebenwirkungen vereinfachten würde: Dass jedes Auswahl-Listen-Feld am rechten Rand ein Symbol hat:
![Download](https://user-images.githubusercontent.com/669783/107566730-851ec080-6be5-11eb-8d86-1fab065cd7ee.png)
Klickt man darauf, öffnet vermehrung in einem neuen Tab das Formular des gewählten Datensatzes.

Ich glaube, das wäre in Vermehrung, wo es so viele solche Felder gibt, generell sehr nützlich. Aber eben, eine Erweiterung.

> Auch wäre es hilfreich (aber auch etwas gefährlich) Zwischenlager, welche auf inaktiv gesetzt werden automatisch zu leeren.

Somit müsste eine Rückfrage erscheinen, welche erst nach ihrer Bestätigung leert. Und es müsste natürlich genau definiert werden, was "leeren" heisst.

**Comment by barbalex** [comment]
Den ersten Fall könnte man auch mit einer Rückfrage abfangen.

**Comment by jmenzi** [comment]
Rückfrage für den ersten Fall finde ich eine gute Idee. Der zweite Fall ist mir allgemein etwas zu heikel. Da nicht prioritär, müssen wir mit der Umsetzung aber noch warten.

**Comment by jmenzi** [comment]
> Ich habe übrigens schon lange eine Idee, die dir das ohne die Nebenwirkungen vereinfachten würde: Dass jedes Auswahl-Listen-Feld am rechten Rand ein Symbol hat:
> ![Download](https://user-images.githubusercontent.com/669783/107566730-851ec080-6be5-11eb-8d86-1fab065cd7ee.png)
> Klickt man darauf, öffnet vermehrung in einem neuen Tab das Formular des gewählten Datensatzes.

Die Idee wäre, dass man eine Kultur für die Anlieferung auswählt, danach erscheint dieses Symbol und man kann draufklicken und das Formular dieser Kultur öffnet sich. Hier könnte man anschliessend die Kultur auf aktiv setzen. Verstehe ich dies richtig?

**Comment by barbalex** [comment]
Genau

**Comment by Seraina1** [comment]
> > Ich habe übrigens schon lange eine Idee, die dir das ohne die Nebenwirkungen vereinfachten würde: Dass jedes Auswahl-Listen-Feld am rechten Rand ein Symbol hat:
> > ![Download](https://user-images.githubusercontent.com/669783/107566730-851ec080-6be5-11eb-8d86-1fab065cd7ee.png)
> > Klickt man darauf, öffnet vermehrung in einem neuen Tab das Formular des gewählten Datensatzes.
> 
> Die Idee wäre, dass man eine Kultur für die Anlieferung auswählt, danach erscheint dieses Symbol und man kann draufklicken und das Formular dieser Kultur öffnet sich. Hier könnte man anschliessend die Kultur auf aktiv setzen. Verstehe ich dies richtig?

Habe mir dieses Issue durchgelesen. Finde diese Idee gut, da man sonst jeweils (etwas kompliziert) nachschauen muss, ob eine (vermeindlich) neue Kultur nicht doch bereits inaktiv besteht. Wäre das auch etwas für das Jahr 2023?


> Auch wäre es hilfreich (aber auch etwas gefährlich) Zwischenlager, welche auf inaktiv gesetzt werden automatisch zu leeren.

Diese Erweiterung finde auch ich allerdings etwas heikel und würde ich vorerst nicht weiter verfolgen.

**Comment by barbalex** [comment]
> Wäre das auch etwas für das Jahr 2023?

Klar, wäre möglich.

**Comment by barbalex** [comment]
Ich bin jetzt nicht ganz sicher, ob wir diesen Issue noch richtig verstehen. Was ist für die Umsetzung vorgeschlagen?

- Der Automatismus, eine Kultur automatisch aktiv zu setzen, wenn in eine inaktive Kultur geliefert wird?
- Die Möglichkeit, in Auswahllisten direkt die gewählte Beziehung in einem neuen Fenster zu öffnen?

**Comment by Seraina1** [comment]
Mir ist nicht ganz klar, was genau du mit dem zweiten Punkt meinst. Meinst du mit Auswahlliste eine Liste mit bestehenden inaktiven und aktiven Kulturen, die in einem neuen Fenster angezeigt werden?

**Comment by barbalex** [comment]
Das mit dem Automatismus ist natürlich machbar. Aber wegen der oben beschriebenen Gefahren müsste die Benutzerführung aufwändig programmiert werden: Die Benutzerin muss explizit gefragt werden, ob die bisher inaktive Kultur aktiv geschaltet werden soll.

So wäre alles Klar und eure Arbeit erleichtert.

**Comment by Seraina1** [comment]
> Soviel ich weiss, werden inaktive Kulturen in der Liste auch angezeigt, wenn man im Menü oben rechts die entsprechende Option wählt. Ev. wäre es hilfreich, wenn sie in der Liste optisch klar als inaktiv erkenntlich wären (abgegraut oder mit "inaktiv" im Namen).
> 
> Dann könnte die Benutzerin sehr bewusst die inaktive Kultur wählen. Und danach mit Klick auf den Link im sich öffnenden Fentser aktiv setzen.

Stimmt, daran habe ich nicht mehr gedacht. Das anzeigen lassen und farbliche abheben der inaktiven Kulturen und Gärten würde dies natürlich vereinfachen und diese Issue hier bereits beheben/erledigen. 
Somit denke ich können wir dieses Issue beheben, indem wir Issue #114 lösen.

**Comment by barbalex** [comment]
Und für den Link, um eine Auswahl in einem neuen Fenster zu öffnen, einen neuen Issue erzeugen? Wenn ihr das möchtet.

**Comment by Seraina1** [comment]
Habe einen neuen Issue erfasst: siehe  #118

**Comment by barbalex** [comment]
Schliesse, da mit anderen Issues umgesetzt
